src/DependencyInjection/Configuration.php

```diff
@@ -739,6 +741,27 @@ private function createCachePluginNode()
                         ->end()
                     ->end()
                 ->end()
+                ->arrayNode('cache_listeners')
+                    ->info('An array of classes to act on the response based on the results of the cache check. Must implement '.CacheListener::class.'. Defaults to an empty array.')
+                    ->beforeNormalization()->castToArray()->ifEmpty()->thenUnset()->end()
+                    ->defaultValue([])
+                    ->prototype('scalar')
+                        ->validate()
+                            ->ifTrue(function ($v) {
+                                $vs = is_array($v) ? $v : (is_null($v) ? [] : [$v]);
+
+                                return empty($vs) || array_reduce($vs, function ($r, $e) {
+                                    return empty($e) || !class_exists($e) || !(new ReflectionClass($e))->implementsInterface(CacheListener::class);
```

Comment:
when these are service names, the checks would need to move to a compiler extension.

Reply:
done